### PR TITLE
fit continuum: fixed to handle the masked arrays.

### PR DESCRIPTION
**Pull request by ntejos**
As stated. I still do not like much representing the flux, sigma attributed using MaskedArray.compressed()...

### CODE REVIEW

**Comment by profxj** [comment]
you can set masking='none'  :)

Merge when this passes the test.

**Comment by profxj** [comment]
I should add that it is better for one to *not* reassign the
internal data of an XSpectrum1D.  Best to create a new object.

File: linetools/spectra/xspectrum1d.py [hunk 1]
```diff
@@ -1301,15 +1304,17 @@ def fit_continuum(self, knots=None, edges=None, wlim=None, dw=10.,
...
                                    contpoints, co=co_init, fig=fig, anchor=anchor)
 
         # wait until the interactive fitting has finished
         while not wrapper.finished:
             plt.waitforbuttonpress()
 
         print('Updating continuum')
-        self.data['co'] = wrapper.continuum
+        # import pdb; pdb.set_trace()
+        gdp = ~self.data['wave'][self.select].mask
```
Comment (ntejos) [inline linetools/spectra/xspectrum1d.py]: I see, so this will crash when `masking='none'`...

Comment (profxj) [inline linetools/spectra/xspectrum1d.py]: I don't think it should.  Will it?

Comment (ntejos) [inline linetools/spectra/xspectrum1d.py]: Not so sure anymore. Will wait until tomorrow morning to give it a second pass before merging then (and to solve the Python 3 error).

**Comment by ntejos** [comment]
Regarding reassignation of internal self.data, how would it work when using the internal method fit_continuum? Would you prefer self.fit_continuum() to return a copy of itself with the continuum?

**Comment by ntejos** [comment]
@profxj Do you mind if I merge?

**Comment by profxj** [comment]
Internal assignment can be done.  I see now that it was simply
being done wrong in that code.  

For continua, you can do:

self.co = xxxx

but if it is the first time you are setting it (as I presume for fit_continuum)
then you should use the normalize() method.

Consider htis then merge.

**Pull request merged**